common/persistence/cassandra/cassandraVisibilityPersistence.go

```diff
@@ -213,6 +218,7 @@ func (v *cassandraVisibilityPersistence) RecordWorkflowExecutionStarted(
 func (v *cassandraVisibilityPersistence) RecordWorkflowExecutionClosed(
 	request *p.RecordWorkflowExecutionClosedRequest) error {
 	batch := v.session.NewBatch(gocql.LoggedBatch)
+	memo, _ := json.Marshal(request.Memo)
```

Comment:
I don't see Marshal called anywhere in the persistence layer. This breaks this. Also anywhere we use marshalling we store encoding.

Reply:
Updated. I refactored history serialization and used same thing for memo